src/firefly/java/edu/caltech/ipac/firefly/server/visualize/VisServerOps.java

```diff
@@ -1434,9 +1437,66 @@ public static WebPlotResult getDS9Region(String fileKey) {
             retval = createError("on getDSRegion", null, e);
         }
         return retval;
+    }
 
+    private static final Map<String, String> footprintMap;
+    static {
```

Comment:
Yes, good point. I would decouple the data content and the loading. The same way we load properties. Actually you could think a future use case of a user-defined footprint to be loaded and added to the layers as part of all the predefined footprint available. 

--
Emmanuel Joliet
NASA/IPAC  Infrared Science Archive
MS 100-22, Caltech, Pasadena, CA 91125
Office: 170
Email:  ejoliet@ipac.caltech.edu
Phone:  (626) 395-1489

> On Jul 18, 2016, at 12:58, Cindy Wang <notifications@github.com> wrote:
> 
> In src/firefly/java/edu/caltech/ipac/firefly/server/visualize/VisServerOps.java:
> 
> >  
> > +    private static final Map<String, String> footprintMap;
> > +    static {
> besides the issue regrading where to put the footprint files, I also have some concern regrading if it is ok to hardcode the footprint map in the java file? or put the map data into a text file and create the map by loading the text file?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>